src/metaopt/io/database/__init__.py

```diff
@@ -148,4 +148,70 @@ class DatabaseError(RuntimeError):
     pass
 
 
+# Include current implementations of the wrapper
 from metaopt.io.database.mongodb import MongoDB  # noqa
+
+
+class DatabaseType(SingletonType):
+    """Instantiate appropriate wrapper for the infrastructure based on input
+    argument, ``dbtype``.
+
+    Attributes
+    ----------
+    all : list of `AbstractDB` classes
+       Updated to contain `all` possible implementations currently. Check out code.
+    typenames : list of str
+       Names of implemented wrapper classes, correspond to possible ``dbtype``
+       values.
+
+    """
+
+    all = [AbstractDB, MongoDB]
```

Comment:
Can't we make AbstractDB register automatically any children? In that case `all` could be replaced by something like `AbstractDB.types`. Lists like `all` are annoying to maintain. Test at line 195 would not be needed anymore.

Reply:
I could make something dirty here:
1. globbing for all Python files in database directory
2. try to import them
3. for each public name in imported module, check whether it is a subclasses of `AbstractDB`,
append to list of `types` if it is.

Reply:
Yeah that's too dirty. :stuck_out_tongue:

Man, I just learned something so nice. :open_mouth: Did you know about ```SomeClass.__subclasses__()```? I just found that from [here](https://stackoverflow.com/a/5883218)

```Python
>>> class A(object):
...     pass
... 
>>> class B(A):
...     pass
... 
>>> A.__subclasses__()
[<class '__main__.B'>]
```

Magic! :tada: